src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -157,6 +154,32 @@ public static TripInstruction alignTravelerToTrip(
         return null;
     }
 
+    /**
+     * Send bus notification if the first leg is a bus leg or approaching a bus leg and within the notify window.
+     */
+    public static boolean sendBusNotification(
+        TravelerPosition travelerPosition,
+        boolean isStartOfTrip,
+        TripStatus tripStatus
+    ) {
+        if (shouldNotifyBusOperator(travelerPosition, isStartOfTrip)) {
+            BusOperatorActions
+                .getDefault()
+                .handleSendNotificationAction(tripStatus, travelerPosition);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Given the traveler's position and leg type, check if bus notification should be sent.
+     */
+    public static boolean shouldNotifyBusOperator(TravelerPosition travelerPosition, boolean isStartOfTrip) {
+        return (isStartOfTrip)
```

Comment:
Extract a variable for the leg to pass to `isBusLeg` and `isWithinOperationalNotifyWindow` depending on `isStartOfTrip`, and remove line 180.